darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterStrategyFactoryImpl.java

```diff
@@ -109,18 +112,38 @@ public DarkClusterStrategy getOrCreate(@Nonnull String darkClusterName, @Nonnull
    */
   private DarkClusterStrategy createStrategy(String darkClusterName, DarkClusterConfig darkClusterConfig)
   {
-    if (darkClusterConfig.hasMultiplier() && darkClusterConfig.getMultiplier() > 0)
-    {
-      BaseDarkClusterDispatcher baseDarkClusterDispatcher = new BaseDarkClusterDispatcherImpl(darkClusterName, _darkClusterDispatcher,
-                                                                                              _notifier, _verifierManager);
-      return new RelativeTrafficMultiplierDarkClusterStrategy(_sourceClusterName, darkClusterName, darkClusterConfig.getMultiplier(),
-                                                              baseDarkClusterDispatcher, _notifier, _facilities.getClusterInfoProvider(), _random);
-    }
-    else
+    if (darkClusterConfig.hasMultiplierStrategyList())
     {
-      // Falling into this clause means that the user does not want traffic to be sent to the dark cluster, temporary or otherwise.
-      return new NoOpDarkClusterStrategy();
+      MultiplierStrategyTypeArray strategyList = darkClusterConfig.getMultiplierStrategyList();
+      for (com.linkedin.d2.multiplierStrategyType multiplierStrategyType : strategyList)
+      {
+        if (RELATIVE_TRAFFIC.equals(multiplierStrategyType) && darkClusterConfig.getMultiplier() > 0)
+        {
+          if (darkClusterConfig.getMultiplier() <= 0)
+          {
+            // it is a valid case to set RELATIVE_TRAFFIC but set the multiplier to zero to
+            // temporarily set the traffic to zero; don't log anything.
+            return new NoOpDarkClusterStrategy();
+          }
+          BaseDarkClusterDispatcher baseDarkClusterDispatcher =
+            new BaseDarkClusterDispatcherImpl(darkClusterName, _darkClusterDispatcher, _notifier, _verifierManager);
+          return new RelativeTrafficMultiplierDarkClusterStrategy(_sourceClusterName, darkClusterName, darkClusterConfig.getMultiplier(),
+                                                                  baseDarkClusterDispatcher, _notifier, _facilities.getClusterInfoProvider(),
+                                                                  _random);
+        }
+        else if (CONSTANT_QPS.equals(multiplierStrategyType))
+        {
+          // the constant qps strategy is not yet implemented, continue to the next strategy if it exists
+        }
+        else
```

Comment:
This else statement return breaks the use of strategy list since it will never iterate past the first element. but also made a previous comment on not using a list.

Reply:
doing this will still use the strategylist, because if constant qps isn't available, it will go into the else if and not the else:
if (RELATIVE_TRAFFIC) {
}
else if (CONSTANT_QPS) {
}
else {
}